src/passes/inheritanceInliner/constructorInheritance.ts

```diff
@@ -0,0 +1,291 @@
+import assert from 'assert';
+import {
+  ContractDefinition,
+  Expression,
+  ExpressionStatement,
+  FunctionDefinition,
+  FunctionKind,
+  FunctionStateMutability,
+  FunctionVisibility,
+  ModifierDefinition,
+  Statement,
+  VariableDeclaration,
+  VariableDeclarationStatement,
+} from 'solc-typed-ast';
+import { AST } from '../../ast/ast';
+import { cloneASTNode } from '../../utils/cloning';
+import { generateFunctionCall } from '../../utils/functionGeneration';
+import { createBlock, createIdentifier, createParameterList } from '../../utils/nodeTemplates';
+import { updateReferencedDeclarations } from './utils';
+
+export function solveConstructorInheritance(
```

Comment:
There are some quite non-trivial transforms going on in this file. There should be a top level comment summarising what's going on. It should explain how there are two different kinds of delegated constructors (modifier-like vs specifier) and at a high level how each is handled. Shorthand code examples of before and after could even be helpful